### PR TITLE
Node Upgrade

### PR DESCRIPTION
**Pull request by SabreCat**
[//]: # (Note: See https://habitica.fandom.com/wiki/Using_Your_Local_Install_to_Modify_Habitica%27s_Website_and_API for more info)

[//]: # (Put Issue # here, if applicable. This will automatically close the issue if your PR is merged in)
Attempts to upgrade NodeJS to latest version. Version 14 which Habitica currently uses is no longer supported.

### CODE REVIEW

File: test/api/v3/integration/challenges/POST-challenge_flag.test.js [hunk 1]
```diff
@@ -41,11 +41,11 @@ describe('POST /challenges/:challengeId/flag', () => {
   });
 
   it('flags a challenge with a higher count when from an admin', async () => {
-    await user.update({ 'contributor.admin': true });
+    const admin = await generateUser({ 'contributor.admin': true });
```
Comment (negue) [inline test/api/v3/integration/challenges/POST-challenge_flag.test.js]: TODO for me - revert this again and see if its working again

might have something to do with the `domain-middleware` that the CI's stopped for it

**Comment by rafucuk** [comment]
What's left with this PR

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -228,15 +237,16 @@ jobs:
...
-    - uses: actions/checkout@v1
+    - uses: actions/checkout@v4
       with:
         fetch-depth: 1
     - name: Use Node.js ${{ matrix.node-version }}
-      uses: actions/setup-node@v1
+      uses: actions/setup-node@v4
       with:
         node-version: ${{ matrix.node-version }}
+    - run: sudo apt-get -y install libkrb5-dev
```
Comment (negue) [inline .github/workflows/test.yml:249]: @SabreCat which program needed this dependency?

File: test/api/v3/integration/challenges/POST-challenge_flag.test.js (hunk 1 above)
Comment (negue) [reply test/api/v3/integration/challenges/POST-challenge_flag.test.js]: will try that again tomorrow

**Comment by negue** [comment]
@SabreCat created #15040 to offload the storybook changes of this PR - could you take a look at maybe just merge - then this one is a bit less cluttered

**Pull request closed without merging**